zokrates_core/src/flatten/mod.rs

```diff
@@ -43,6 +106,258 @@ impl<'ast, T: Field> Flattener<'ast, T> {
+        condition: BooleanExpression<'ast, T>,
+        consequence: U,
+        alternative: U,
+    ) -> Vec<FlatExpression<T>> {
+        let condition = self.flatten_boolean_expression(symbols, statements_flattened, condition);
+
+        let consequence = consequence.flatten(self, symbols, statements_flattened);
+
+        let alternative = alternative.flatten(self, symbols, statements_flattened);
+
+        let size = consequence.len();
+
+        let condition_id = self.use_sym();
+        statements_flattened.push(FlatStatement::Definition(condition_id, condition));
+
+        let consequence_ids: Vec<_> = (0..size).map(|_| self.use_sym()).collect();
+        statements_flattened.extend(
+            consequence
+                .into_iter()
+                .zip(consequence_ids.iter())
+                .map(|(c, c_id)| FlatStatement::Definition(*c_id, c)),
+        );
+
+        let alternative_ids: Vec<_> = (0..size).map(|_| self.use_sym()).collect();
+        statements_flattened.extend(
+            alternative
+                .into_iter()
+                .zip(alternative_ids.iter())
+                .map(|(a, a_id)| FlatStatement::Definition(*a_id, a)),
+        );
+
+        let term0_ids: Vec<_> = (0..size).map(|_| self.use_sym()).collect();
+        statements_flattened.extend(consequence_ids.iter().zip(term0_ids.iter()).map(
+            |(c_id, t0_id)| {
+                FlatStatement::Definition(
+                    *t0_id,
+                    FlatExpression::Mult(
+                        box condition_id.clone().into(),
+                        box FlatExpression::from(*c_id),
+                    ),
+                )
+            },
+        ));
+
+        let term1_ids: Vec<_> = (0..size).map(|_| self.use_sym()).collect();
+        statements_flattened.extend(alternative_ids.iter().zip(term1_ids.iter()).map(
+            |(a_id, t1_id)| {
+                FlatStatement::Definition(
+                    *t1_id,
+                    FlatExpression::Mult(
+                        box FlatExpression::Sub(
+                            box FlatExpression::Number(T::one()),
+                            box condition_id.into(),
+                        ),
+                        box FlatExpression::from(*a_id),
+                    ),
+                )
+            },
+        ));
+
+        let res: Vec<_> = (0..size).map(|_| self.use_sym()).collect();
+        statements_flattened.extend(term0_ids.iter().zip(term1_ids).zip(res.iter()).map(
+            |((t0_id, t1_id), r_id)| {
+                FlatStatement::Definition(
+                    *r_id,
+                    FlatExpression::Add(
+                        box FlatExpression::from(*t0_id),
+                        box FlatExpression::from(t1_id),
+                    ),
+                )
+            },
+        ));
+        res.into_iter().map(|r| r.into()).collect()
+    }
+
+    /// Flatten a array selection expression
```

Comment:
"an", if you wanna be nitpicky ;)

Reply:
wanna try the github "suggest change" feature? :p